php/Terminus/Models/Collections/TerminusCollection.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Terminus\Models\Collections;
+
+use \TerminusCommand;
+
+abstract class TerminusCollection {
+  protected $fetch_args = array();
+  protected $fetch_url  = '';
+  protected $models     = array();
+  protected $owner_name = false;
+
+  /**
+   * Instantiates the collection, sets param members as properties
+   *
+   * @param [array] $options To be set to $this->key
+   * @return [TerminusCollection] $this
+   */
+  public function __construct($options = array()) {
+    foreach ($options as $key => $option) {
+      $this->$key = $option;
+    }
+  }
+
+  /**
+   * Retrieves all models
+   *
+   * @return [array] $this->models
+   */
+  public function all() {
+    if (empty($this->models)) {
+      $this->fetch();
+    }
+    return $this->models;
+  }
+
+  /**
+   * Fetches model data from API and instantiates their model instances
+   *
+   * @return [void]
+   */
+  public function fetch() {
+    $model   = $this->getModelName();
+    $owner   = $this->owner_name;
+    $options = array_merge(
+      array('options' => array('method' => 'get')),
+      $this->fetch_args
+    );
+    $results = TerminusCommand::simple_request(
+      $this->fetch_url,
+      $options
+    );
+
+    foreach (get_object_vars($results['data']) as $id => $model_data) {
```

Comment:
More as an FYI: within Backbone.js's fetch, it also abstracts out methods of "parse" (for re-arranging the api's response data if it needs it...for example, some pantheon api's return a hash of objects keyed off their ID, and others return a proper collection list) and an "add" method that is reponsible for taking a model and adding it to the collection (and setting any of the collection's properties like owner, etc on it).